src/renderer/base/renderer.hpp

```diff
@@ -46,6 +46,7 @@ namespace Microsoft::Console::Render
         virtual ~Renderer() override;
 
         [[nodiscard]] HRESULT PaintFrame();
+        void TriggerIntereaction(IntereactionType type) override;

```

Comment:
Personally I really want this to be `TriggerIntereaction(IntereactionType type, void* pData)`. 

Right now I can only use this for methods without parameters (for example `TriggerSelection`). With additional `void *` parameter I can also use it for `TriggerRedrawCursor(const COORD* const pcoord)` and `TriggerScroll(const COORD* const pcoordDelta)`, just by casting the `void *` to specific type.

Obviously using `void *` means losing the compile-time safety for this method. So I'm not really sure.

Reply:
I'm not certain what the benefit is of funneling all of the interaction types through a single function. Reducing the _number_ of virtual functions in an interface is not going to help with the _cost of calling a single virtual function_, right?

Reply:
Well, because @lhecker and I want to generally simplify the IRenderEngine & IRenderTarget interface so that we can kill them in the future. Don’t tell them, it’ll break their heart😸

Reply:
You're just replacing four slots in the vtable with one slot and a big `switch(...)`. I just.. I don't think it's the right way to simplify the interface, is it?

Reply:
Technically yes. I am replacing vttable with switch. The idea behind this is IRenderEngine should expose as less methods as possible to increase the encapsulation, and let the child class decides why it can do with all kinds of intereaction. Currently There’s a lot of empty implementation for TriggerSomething in renderers that do even care about user intereaction. This seem to indicate exposing (and forcefully overriding) too many TriggerSomething is flawed architecturally.

Reply:
I am open for ideas! Please enlighten me, everyone 😋

Reply:
@skyline75489 I can't speak to the `TriggerSomething` architectural flaw.  However creating a case of what seems like control coupling does not seem to be a good way to cure it.  Maybe there are some sort of missing factoring and composition cases that improves the abstraction and encapsulation.  Hiding semantic-relevant decisions does not smell like good encapsulation, if that is what's ha;ppening here.  (I will now collapse back into my arm chiar.)

Reply:
Thanks @DHowett and @orcmid . I now figured that I don't like this, either. I'll revert my last commit and try something else.